src/module/vue/components/progress/progress-list-item.vue

```diff
@@ -0,0 +1,238 @@
+        @click="toggleComplete"
+        :tooltip="completedTooltip"
+      />
+      <BtnFaicon
+        class="block"
+        v-if="editMode && item.data.hasTrack"
+        icon="caret-left"
+        @click="retreat"
+        :tooltip="$t('IRONSWORN.UnmarkProgress')"
+      />
+      <BtnFaicon
+        class="block"
+        v-if="item.data.hasTrack"
+        icon="caret-right"
+        @click="advance"
+        :tooltip="$t('IRONSWORN.MarkProgress')"
+      />
+      <BtnRollprogress
+        v-if="item.data.hasTrack"
+        :item="item"
+        :tooltip="$t('IRONSWORN.ProgressRoll')"
+        class="block"
+      />
+      <BtnFaicon
+        v-if="showStar"
+        class="star-progress block"
+        icon="star"
+        :solid="item.data.starred"
+        :tooltip="$t('IRONSWORN.StarProgress')"
+        data-tooltip-direction="RIGHT"
+        @click="toggleStar"
+      />
+    </section>
+  </article>
+</template>
+
+<style lang="less" scoped>
+@progress_widget_spacing: 6px;
+.progress-list-item {
+  padding: (@progress_widget_spacing / 2) (@progress_widget_spacing / 2)
+    @progress_widget_spacing;
+  display: grid;
+  grid-template-columns: max-content max-content 1fr max-content;
+  grid-template-rows: max-content max-content 1fr;
+  gap: @progress_widget_spacing;
+  .progress-img {
+    grid-column: 2;
+    grid-row: 1 / span 2;
+    margin: 0;
+  }
+  .progress-rank-pips {
+    grid-column: 3 / span 2;
+    grid-row: 1;
+  }
+  .progress-title {
+    grid-column: 3;
+    grid-row: 2;
+    margin: 0;
+    line-height: 1;
+    height: max-content;
+  }
+  .progress-subtitle {
+    grid-column: 1;
+    grid-row: 1 / span 3;
+    padding: 0;
+    line-height: 1;
+    width: max-content;
+    padding: 0;
+    margin: 0;
+    font-weight: normal;
+  }
+  .progress-widgets {
+    grid-column: 2 / span 3;
+    grid-row: 3;
+    gap: 2px;
+    .progress-clock {
+      flex-basis: 50px;
+    }
+  }
+  .progress-controls {
+    display: grid;
+    grid-column: 4;
+    grid-row: 1 / span 2;
+    grid-auto-flow: column;
+    > * {
+      aspect-ratio: 1;
+      grid-row: 1;
+    }
+    .star-progress {
+      grid-row: 2;
+    }
+  }
+}
+</style>
+
+<script lang="ts" setup>
+import { capitalize, computed, inject, provide } from 'vue'
+import { $ActorKey } from '../../provisions'
+import Clock from '../clock.vue'
+import Track from './track.vue'
+import BtnRollprogress from '../buttons/btn-rollprogress.vue'
+import BtnFaicon from '../buttons/btn-faicon.vue'
+import RankPips from '../rank-pips/rank-pips.vue'
+import DocumentImg from '../document-img.vue'
+import { RANKS } from '../../../constants.js'
+import ProgressTrack from './progress-track.vue'
+
+const props = defineProps<{
+  item: any
+  actor: any
+  showStar?: boolean
+  /**
+   * When true, renders the progress bar for more compact display.
+   */
+  compactProgress?: boolean
+}>()
+
+const $actor = inject($ActorKey)
+provide(
+  'actor',
+  computed(() => props.actor)
+)
```

Comment:
Here you're passing actor as a prop, but you're also injecting it. Less repetition if you lose the prop and just use the injected value:

```suggestion
const $actor = inject($ActorKey)
provide(
  'actor',
  computed(() => $actor?.toObject())
)
```

Reply:
Heck, it's possible that `actor` is already being provided here. They usually come as a pair.

Just to clarify, these are the type norms for provided things:

- `$actor` via `$ActorKey` — instance of a Foundry `IronswornActor` class
- `actor` via `'actor'` — the output of something like `actor.toObject()`
- (similar for `$item` and `item`)

Reply:
TBH, i monkeyed with this for several hours before deciding "i guess this works, fuck it" and giving up. juggling various elements with fairly abstract names like `$actor`, `$item`, `actor`, `item`, and `$actorItem` was frustrating (and, yes, potentially discernable from context, but making it an 'it's fine if you just remember...' is a big 'if', at least for me). but, not being 100% clear on their relationship/intended purposes in the first place, i'm not well-equipped to resolve it by renaming them, either.

Reply:
Yeah, I understand that feeling. As a rule of thumb, I think you can mostly remove any `provide` calls that you've added in this PR, since they're usually re-providing something that's already being provided in a parent node.

Maybe someday I'll write a developers' guide to this system.